lib/matplotlib/backends/backend_qt5.py

```diff
@@ -1123,27 +1160,25 @@ class _BackendQT5(_Backend):
     def trigger_manager_draw(manager):
         manager.canvas.draw_idle()
 
-    @staticmethod
-    def interrupt_handler(*args):
-        pyplot.close()
-        _BackendQT5.interrupted = True
-
     @staticmethod
     def mainloop():
         old_signal = signal.getsignal(signal.SIGINT)
         # allow SIGINT exceptions to close the plot window.
-        signal.signal(signal.SIGINT, _BackendQT5.interrupt_handler)
-        timer = QtCore.QTimer()
-        timer.start(500)  # every 0.5s we are able to catch a SIG_INT
-                          # in the interpreter and it then will be handled
-                          # by _BackendQT5.interrupt_handler
-        timer.timeout.connect(lambda: None)
+
+        SIGINTHandler(qApp)
```

Comment:
This may theoretically be garbage-collected immediately. Better to use a context manager.

Reply:
I guess it wasn't because it was a QObj had a parent qApp, but now it's not relevant